src/main/java/com/google/devtools/build/lib/remote/RemoteModule.java

```diff
@@ -336,6 +376,17 @@ public void beforeCommand(CommandEnvironment env) throws AbruptExitException {
             RemoteActionContextProvider.createForRemoteCaching(
                 env, remoteCache, retryScheduler, digestUtil);
       }
+
+      if (enableRemoteDownloader) {
+        remoteDownloaderSupplier.set(new GrpcRemoteDownloader(
+            downloaderChannel.retain(),
+            Optional.ofNullable(credentials),
+            retrier,
+            repoContext,
+            cacheClient,
+            remoteOptions));
+        downloaderChannel.release();
```

Comment:
Why the immediate release after the retain?

Reply:
Trying to match the existing code. Ctrl+F `new ByteStreamUploader(` or `new GrpcRemoteExecutor(` for where I'm looking as examples.

Reply:
Thanks for the explanation. I'll import this today and try to cut a new RC of 3.0.0.